options.go

```diff
@@ -177,8 +197,38 @@ func IdentityResolverFunc(resolver IdentityResolver) Option {
 	}
 }
 
+// RingChecksumStatPeriod configures the period between emissions of the stat
+// 'ring.checksum-periodic'. Using a value <=0 (or RingChecksumStatPeriodNever)
+// will disable emission of this stat. Using a value in (0, 10ms) will return
+// an error, as that value is unrealistically small. Normal values must
+// therefore be >=10ms. RingChecksumStatPeriodDefault defines the default.
+func RingChecksumStatPeriod(period time.Duration) Option {
+	return func(r *Ringpop) error {
+		if period <= 0 {
+			period = RingChecksumStatPeriodNever
+		} else if period < 10*time.Millisecond {
+			return errors.New("RingChecksumStatPeriod invalid below 10 ms")
+		}
+		r.config.RingChecksumStatPeriod = period
+		return nil
+	}
+}
+
 // Default options
 
+// RingChecksumStatPeriodNever defines a "period" which disables
+// ring.checksum-periodic stat emission.
+const RingChecksumStatPeriodNever = time.Duration(-1)
```

Comment:
@dansimau There is no current notion of 'stats' across the board. It might be sensible to create one, but I went more minimal.

@CorgiMan Also possible, but then it's no longer a pass through from a config file. Do we have a pattern/example for this type of thing? I would rather follow the existing pattern then invent my own...